### DISCUSSION TITLE
Porting dyn4j to web

### DISCUSSION THREAD

**Discussion by maheshkurmi** [dyn4j/dyn4j#206] [Ideas]
I want to use dyn4j in web based application using GWT? 
Is there any module in dyn4j which is not gwt compatible ? Share feedback if somebody already tried doing so.
I think making dyn4j available to web will be awesome.

**Comment by wnbittle** [comment]
I've never used GWT, but comparing [this link](http://www.gwtproject.org/doc/latest/RefJreEmulation.html#Package_java_lang) to a cursory search of dyn4j only revealed `ResourceBundle` as an issue.  dyn4j makes heavy use of java.util, particularly the collections framework, all of which appear to be supported.

In dyn4j the class that uses `ResourceBundle` is `org.dyn4.resources.Messages` which could be removed in a fork if you didn't need _translated_ error messages.  I'm temped to remove this anyway since I don't have a way to maintain translations.  All the messages are for the developer writing code against dyn4j so could probably be in English only.

The JUnit tests _might_ work - there's some [tutorial on JUnit testing](http://www.gwtproject.org/doc/latest/tutorial/JUnit.html).  The samples projects definitely won't work as they rely on Java 2D and some 1.6 features like `System.nanoTime()`.

**Reply by maheshkurmi** [reply]
Thanks William,
I will try, will share once I create a working prototype.
